src/main/kotlin/com/coder/toolbox/util/URLExtensions.kt

```diff
@@ -13,6 +13,20 @@ fun URL.withPath(path: String): URL = URL(
     if (path.startsWith("/")) path else "/$path",
 )
 
+fun URL.withLastSegment(segment: String): URL {
```

Comment:
This looks essentially like it is resolving a path relatively, so simply `URL(url, segment)` would also work.  Or `url.toURI().resolve(segment).toURL()`.

Reply:
Nice catch, thx!